viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(movie_title, genre, rating):
+    if movie_title == None or genre == None or rating == None:
+        return None
+    movie = {"title":movie_title, "genre":genre, "rating":rating}
+    return movie
 
+def add_to_watched(user_data, movie):
+    if movie["title"] not in user_data["watched"]:
+        user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    if movie["title"] not in user_data["watchlist"]:
+        user_data["watchlist"].append(movie)
+    return user_data
+    
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"] and movie_title not in user_data["watched"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = 0.0
+    count = 0
+
+    if not user_data:
+        return None
+
+    for movies in user_data.values():

```

Comment:
The tests in Wave 2 arrange `janes_data` which happens to be a data structure that looks something like:
```py
{'watched': [{...}, {...}, {...}, {...}, {...}, {...}]}
``` 
So it does work out that you can directly use `values`. What if the `user_data` also has keys and values related to watchlist, friends, etc? Something like:
```py
{'watched': [{...}, {...}, {...}, {...}, {...}, {...}], 'watchlist': [], 'friends': []}
```

Then when you use `values`, you'll be looping over ALL the values not just the values associated with 'watched', and you'll see a bug has been created in your code. 

It is important to be **explicit** about what you want to loop over and not make assumptions about this based only on a specific subset of arranged data in a single unit test. What exactly do we want to loop over? We want to loop over only the movies a user has watched, so this loop should look like this:

```suggestion
    for movie in user_data["watched]:
```